src/ucs/sys/sys.c

```diff
@@ -744,6 +744,157 @@ ucs_status_t ucs_tcpip_socket_create(int *fd_p)
     return UCS_OK;
 }
 
+ucs_status_t ucs_socket_connect(int fd, const struct sockaddr *dest_addr)
+{
+    char str[UCS_SOCKADDR_STRING_LEN];
+    int ret;
+
+    ret = connect(fd, dest_addr, ucs_sockaddr_sizeof(dest_addr));
+    if (ret < 0) {
+        if (errno == EINPROGRESS) {
+            ucs_debug("connect(fd=%d, dest_addr=%s): %m", fd,
+                      ucs_sockaddr_str(dest_addr, str, UCS_SOCKADDR_STRING_LEN));
+            return UCS_INPROGRESS;
+        }
+
+        ucs_error("connect(fd=%d, dest_addr=%s) failed: %m", fd,
+                  ucs_sockaddr_str(dest_addr, str, UCS_SOCKADDR_STRING_LEN));
+        return UCS_ERR_UNREACHABLE;
+    }
+
+    return UCS_OK;
+}
+
+ucs_status_t ucs_socket_connect_nb_get_status(int fd)
+{
+    socklen_t conn_status_sz;
+    int ret, conn_status;
+
+    conn_status_sz = sizeof(conn_status);
+
+    ret = getsockopt(fd, SOL_SOCKET, SO_ERROR,
+                     &conn_status, &conn_status_sz);
+    if (ret < 0) {
+        ucs_error("getsockopt(fd=%d) failed to get SOL_SOCKET(SO_ERROR): %m",
+                  fd);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (conn_status == EINPROGRESS || conn_status == EWOULDBLOCK) {
+        return UCS_INPROGRESS;
+    }
+
+    if (conn_status != 0) {
+        ucs_error("SOL_SOCKET(SO_ERROR) status on fd %d: %d", fd, conn_status);
+        return UCS_ERR_UNREACHABLE;
+    }
+
+    return UCS_OK;
+}
+
+ucs_status_t ucs_socketpair_create(int domain, int type, int *fd_1_p, int *fd_2_p)
```

Comment:
no need to use UNIX domain sockets in TCP, this is handled by self/shm transports